File: Proceso/Lab/VisualBasicTFS/AccesoDatos/Conexion.vb
Imports System.Data.SqlClient

Public Class Conexion
    Private connection As SqlConnection
    Private connStr As String
    Private command As SqlCommand
    ' Constructor
    Public Sub New()
        Me.connStr = "Data Source=.;Initial Catalog=Laboratorio1Proceso;Integrated Security=True"
    End Sub
    ' Inicializar
    Public Function Bootstrap() As Conexion
        Try
            Me.connection = New SqlConnection(Me.connStr)
            Me.connection.Open()
            Return Me
        Catch ex As Exception
            Return Nothing
        End Try
    End Function
    ' Finalizacion
    Public Function CloseConnection() As Boolean
        Try
            Me.connection.Close()
            Me.connection.Dispose()
            Return True
        Catch ex As Exception
            Return False
        End Try
    End Function
    ' Agregar SQL Query
    Public Function SQLQuery(ByVal commStr) As Conexion
        Try
            Me.command = New SqlCommand(commStr, Me.connection)
            Return Me
        Catch ex As Exception
            Return Nothing
        End Try
    End Function
    Public Function IsStoredProcedure() As Conexion
        Try
            Me.command.CommandType = CommandType.StoredProcedure
            Return Me
        Catch ex As Exception
            Return Nothing
        End Try
    End Function
    ' Agregar Parametros
    Public Function SQLParam(ByVal _name As String, ByVal _value As Object, ByVal _type As SqlDbType, Optional ByVal _size As Integer = Nothing) As Boolean
        Try
            Dim parameter As SqlParameter
            If (_size <> Nothing) Then
                parameter = Me.command.Parameters.Add(_name, _type, _size)
            Else
                parameter = Me.command.Parameters.Add(_name, _type)
            End If
            parameter.Value = _value
            Return True
        Catch ex As Exception
            Return False
        End Try
    End Function
    ' Execute
    Public Function ExecuteStoredProcedure() As SqlDataReader
        Try
            Dim reader As SqlDataReader
            Me.command.Prepare()
            reader = Me.command.ExecuteReader()
            command.CommandTimeout = 0
            Return reader
        Catch ex As Exception
            Return Nothing
        End Try
    End Function
End Class

File: Tecnico/Lab/Lab 2/Lab_2/AccesoDatos/Conexion.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Data;
using System.Data.SqlClient;


public class Conexion
{
    private SqlConnection connection;
    private string connStr;
    private SqlCommand command;
    public Conexion()
    {
        this.connStr = "Data Source=.;Initial Catalog=Laboratorio2;Integrated Security=True";
    }

    public Conexion Bootstrap()
    {
        try
        {
            this.connection = new SqlConnection(this.connStr);
            this.connection.Open();
            return this;
        }
        catch (Exception ex)
        {
            return null;
        }
    }

    public bool CloseConnection()
    {
        try
        {
            this.connection.Close();
            this.connection.Dispose();
            return true;
        }
        catch (Exception ex)
        {
            return false;
        }
    }

    public Conexion SQLQuery(string commStr)
    {
        try
        {
            this.command = new SqlCommand(commStr, this.connection);
            return this;
        }
        catch (Exception ex)
        {
            return null;
        }
    }

    public Conexion IsStoredProcedure()
    {
        try
        {
            this.command.CommandType = CommandType.StoredProcedure;
            return this;
        }
        catch (Exception ex)
        {
            return null;
        }
    }

    public bool SQLParam(string _name, object _value, SqlDbType _type, int _size = 0)
    {
        try
        {
            SqlParameter parameter;
            if ((_size != 0))
            {
                parameter = this.command.Parameters.Add(_name, _type, _size);
            }
            else
            {
                parameter = this.command.Parameters.Add(_name, _type);
            }

            parameter.Value = _value;
            return true;
        }
        catch (Exception ex)
        {
            return false;
        }
    }

    public SqlDataReader ExecuteStoredProcedure()
    {
        try
        {
            SqlDataReader reader;
            this.command.Prepare();
            reader = this.command.ExecuteReader();
            command.CommandTimeout = 0;
            return reader;
        }
        catch (Exception ex)
        {
            return null;
        }
    }
}